tools/dev-doctor/iptablescheck.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+)
+
+// An iptablesRuleCheck checks that the give iptables rule is present.
+type iptablesRuleCheck struct {
+	rule []string
+}
+
+func (c *iptablesRuleCheck) id() string {
+	return "iptables-rule"
+}
+
+func (c *iptablesRuleCheck) run() (checkResult, string) {
+	if runtime.GOOS != "linux" {
+		return checkSkipped, "iptables only used on linux"
+	}
+
+	var cmd *exec.Cmd
+	if os.Getuid() == 0 {
+		cmd = exec.Command("iptables", append([]string{"-C"}, []string(c.rule)...)...)
+		cmd.Stdin = nil
+		cmd.Stdout = nil
+		cmd.Stderr = nil
+
+	} else {
+		cmd = exec.Command("sudo", append([]string{"iptables", "-C"}, []string(c.rule)...)...)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = nil
+		cmd.Stderr = os.Stderr
```

Comment:
Why is stderr printed when running as an unprivileged users but not when running as root? I'm getting the following warnings which I would expect to be silenced since we already print a message for missing iptables rules.
```
iptables: Bad rule (does a matching rule exist in that chain?).
iptables: Bad rule (does a matching rule exist in that chain?).
iptables: Bad rule (does a matching rule exist in that chain?).
```

Reply:
This is because I thought I needed to pass through stderr so that `sudo` could prompt for a password. Fixed now.